cells/std/cli/randomitems.go

```diff
@@ -0,0 +1,273 @@
+		shuf := func(x []string) {
+			rand.Shuffle(len(x), func(i, j int) { x[i], x[j] = x[j], x[i] })
+		}
+		shuf(r.names)
+		shuf(r.cells)
+		shuf(r.organelles)
+		shuf(r.clades)
+		shuf(r.descs)
+	})
+}
+
+func (r *randomItemGenerator) next() item {
+	if r.mtx == nil {
+		r.reset()
+	}
+
+	r.mtx.Lock()
+	defer r.mtx.Unlock()
+
+	var (
+		actionsGenerator randomActionGenerator
+	)
+	// Make actions
+	const numItems = 3
+	items := make([]action, numItems)
+	for i := 0; i < numItems; i++ {
+		items[i] = actionsGenerator.next()
+	}
+
+	i := item{
+		StdName:        r.names[r.nameIndex],
+		StdOrganelle:   r.organelles[r.organelleIndex],
+		StdCell:        r.cells[r.cellIndex],
+		StdClade:       r.clades[r.cladeIndex],
+		StdDescription: r.descs[r.descIndex],
+		actions:        items,
+	}
+
+	r.nameIndex++
+	if r.nameIndex >= len(r.names) {
+		r.nameIndex = 0
+	}
+
+	r.organelleIndex++
+	if r.organelleIndex >= len(r.organelles) {
+		r.organelleIndex = 0
+	}
+
+	r.cellIndex++
+	if r.cellIndex >= len(r.cells) {
+		r.cellIndex = 0
+	}
+
+	r.cladeIndex++
+	if r.cladeIndex >= len(r.clades) {
+		r.cladeIndex = 0
+	}
+
+	r.descIndex++
+	if r.descIndex >= len(r.descs) {
+		r.descIndex = 0
+	}
+
+	return i
+}
+
+type randomActionGenerator struct {
+	actionNames        []string
+	actionNameIndex    int
+	actionCommands     [][]string
+	actionCommandIndex int
+	actionDescs        []string
+	actionDescIndex    int
+	mtx                *sync.Mutex
+	shuffle            *sync.Once
+}
+
+func (r *randomActionGenerator) reset() {
+	r.mtx = &sync.Mutex{}
+	r.shuffle = &sync.Once{}
+	r.actionNames = []string{
+		"build",
+		"run",
+		"deploy",
+		"serve",
+		"validate",
+		"test",
+	}
+	r.actionCommands = [][]string{
+		{"nix", "run", ".#f.r.a.g.m.e.n.t"},
+		{"nix", "build", ".#fragment", "&&", "nomad", "result/job"},
+		{"cat", "./cow"},
+		{"cowsay", "hi"},
+		{"fastlane", "run", "..."},
+		{"go", "build", "."},
+	}
+	r.actionDescs = []string{
+		"A little weird",
+		"Bold flavor",
+		"Can’t get enough",
+		"Delectable",
+		"Expensive",
+		"Expired",
+		"Exquisite",
+		"Fresh",
+		"Gimme",
+		"In season",
+		"Kind of spicy",
+		"Looks fresh",
+		"Looks good to me",
+		"Maybe not",
+		"My favorite",
+		"Oh my",
+		"On sale",
+		"Organic",
+		"Questionable",
+		"Really fresh",
+		"Refreshing",
+		"Salty",
+		"Scrumptious",
+		"Delectable",
+		"Slightly sweet",
+		"Smells great",
+		"Tasty",
+		"Too ripe",
+		"At last",
+		"What?",
+		"Wow",
+		"Yum",
+		"Maybe",
+		"Sure, why not?",
+	}
+
+	r.shuffle.Do(func() {
+		shufStrings := func(x []string) {
+			rand.Shuffle(len(x), func(i, j int) { x[i], x[j] = x[j], x[i] })
+		}
+		shufLists := func(x [][]string) {
+			rand.Shuffle(len(x), func(i, j int) { x[i], x[j] = x[j], x[i] })
+		}
+		shufStrings(r.actionNames)
+		shufLists(r.actionCommands)
+		shufStrings(r.actionDescs)
+	})
+
+}
+
+func (r *randomActionGenerator) next() action {
+	if r.mtx == nil {
+		r.reset()
+	}
+
+	r.mtx.Lock()
+	defer r.mtx.Unlock()
```

Comment:
And also I saw only one place where these generators are called and it's very not threaded, so I don't think locks are needed here.